src/robotFactory.js

```diff
@@ -1,10 +1,75 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords = {}, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = coords;
+    this.coords.x = coords.x || 0;
+    this.coords.y = coords.y || 0;
+    this.chipVersion = chipVersion;
+  }
```

Comment:
You can create coords object in constructor, not like a default value of coords.

```suggestion
  constructor(name, weight, coords , chipVersion) {
    this.name = name;
    this.weight = weight;
    this.coords = {
      x: coords.x || 0;
      y: coords.y || 0;
    };
    this.chipVersion = chipVersion;
  }
```